### PR TITLE
index.js and test.js updated to incorporate testing

### PR DESCRIPTION
**Pull request by Tang-1996**
Moved basic input validation functions into 'module.exports' so that they can be used in the test.js file. Added some simple tests ensuring RegEx only allow valid input.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -151,3 +139,30 @@ function reportLocationUpdateError(reason, callback) {
         })
     });
 }
+
+module.exports = {
```
Comment (danarth) [inline index.js:143]: This `module.exports` section I have not seen before. @Tang-1996 Did you follow a tutorial that told you to do this? If so, please can you reply with a link?

File: test.js [hunk 2]
```diff
@@ -7,13 +7,89 @@ Travis CI runs all of the tests in the /tests directory when building.
...
+describe('Input validation tests', function() {
+    describe('UserID input validation tests', function() {
+      it('UserID test - testing a random 16 character ID', function() {
+        assert.equal(true, myCode.userIdRegExValid("abcdef1111111111"));
+      });
+      it('UserID test - testing a random 10 character ID', function() {
+        assert.equal(false, myCode.userIdRegExValid("abcd234517"));
+      });
+      //test fails---------? Fix Regex
+      it('UserID test - testing a random 17 character ID', function() {
```
Comment (danarth) [inline test.js:21]: According to the Travis build, this test fails:

    AssertionError [ERR_ASSERTION]: false == true
      + expected - actual

This test should be amended (or the initial code in `index.js` should be fixed - @harrygcollins please can you investigate?) and then the Travis build should pass.

**Review by danarth — CHANGES_REQUESTED** [review]
I have added line comments to show where the Travis build failed. Once these are fixed, re-open the pull request and I will take a look.

As explained in the code review, this might not be an issue with tests but maybe the User ID function. You might need @harrygcollins help as I believe he is the author of that function.

File: test.js (hunk 2 above)
Comment (Tang-1996) [reply test.js:21]: From my RegEx experience, the {16} indicated a UserID with at most 16 characters. Given as input (for the test) is a 19 character ID which should not be allowed. @harrygcollins can you please investigate

Comment (Tang-1996) [reply test.js:21]: managed to fix the RegEx

**Pull request closed without merging**